File: spring-boot-oauth2-authorization-server-extended/spring-boot-oauth2-authorization-server-extended-frontend/src/main/java/com/onlyteo/sandbox/controller/ViewController.java
package com.onlyteo.sandbox.controller;

import org.springframework.security.core.Authentication;
import org.springframework.security.oauth2.core.oidc.user.OidcUser;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;

@Controller
public class ViewController {

    @GetMapping(path = "/")
    public String getIndexPage() {
        return "index";
    }

    @ModelAttribute
    public void getModel(final Authentication authentication,
                         final Model model) {
        if (authentication != null && authentication.getPrincipal() instanceof OidcUser principal) {
            model.addAttribute("profile", principal);
        }
    }
}

File: spring-boot-oauth2-authorization-server-extended/spring-boot-oauth2-authorization-server-extended-frontend/src/main/kotlin/com/onlyteo/sandbox/controller/ViewController.kt
package com.onlyteo.sandbox.controller

import org.springframework.security.core.Authentication
import org.springframework.security.oauth2.core.oidc.user.OidcUser
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ModelAttribute

@Controller
class ViewController {

    @GetMapping(path = ["/"])
    fun getIndexPage(): String {
        return "index"
    }

    @ModelAttribute
    fun getModel(
        authentication: Authentication?,
        model: Model
    ) {
        if (authentication != null && authentication.principal is OidcUser) {
            model.addAttribute("profile", authentication.principal)
        }
    }
}